app/uk/gov/hmrc/apisubscriptionfields/service/SubscriptionFieldsService.scala

```diff
@@ -64,22 +69,27 @@ class RepositoryFedSubscriptionFieldsService @Inject()(repository: SubscriptionF
   }
 
   def delete(identifier: SubscriptionIdentifier): Future[Boolean] = {
-    repository.delete(identifier.encode())
+    val id = identifier.encode()
+    logger.debug(s"[delete] SubscriptionIdentifier: $identifier id: $id")
+    repository.delete(id)
   }
 
   def get(identifier: SubscriptionIdentifier): Future[Option[SubscriptionFieldsResponse]] = {
+    logger.debug(s"[get] SubscriptionIdentifier: $identifier")
     for {
       fetch <- repository.fetchById(identifier.encode())
     } yield fetch.map(asResponse)
   }
 
   def get(subscriptionFieldsId: SubscriptionFieldsId): Future[Option[SubscriptionFieldsResponse]] = {
+    logger.debug(s"[get] SubscriptionFieldsId: $subscriptionFieldsId")
     for {
       fetch <- repository.fetchByFieldsId(subscriptionFieldsId.value)
     } yield fetch.map(asResponse)
   }
 
   private def save(apiSubscription: SubscriptionFields): Future[SubscriptionFieldsResponse] = {
+    logger.debug(s"[save] SubscriptionFields: ${apiSubscription.id} ${apiSubscription.fieldsId}")
```

Comment:
Why not writing directly?
```
logger.debug(s"[save] SubscriptionFields: ${apiSubscription}")
```

The parameter of type `SubscriptionFields` should be printed and show you all inner fields by calling internally the `toString`.

`SubscriptionFields` is a case class.
It has the `toString` method automatically given for free.

Reply:
done